src/cinder/Triangulate.cpp

```diff
@@ -108,22 +108,87 @@ void Triangulator::addPolyLine( const vec2 *points, size_t numPoints )
 TriMesh Triangulator::calcMesh( Winding winding )
 {
 	TriMesh result( TriMesh::Format().positions( 2 ) );
-	
+
 	tessTesselate( mTess.get(), (int)winding, TESS_POLYGONS, 3, 2, 0 );
 	result.appendPositions( (vec2*)tessGetVertices( mTess.get() ), tessGetVertexCount( mTess.get() ) );
 	result.appendIndices( (uint32_t*)( tessGetElements( mTess.get() ) ), tessGetElementCount( mTess.get() ) * 3 );
-	
+
 	return result;
 }
 
 TriMeshRef Triangulator::createMesh( Winding winding )
 {
 	TriMeshRef result = make_shared<TriMesh>( TriMesh::Format().positions( 2 ) );
-	
+
 	tessTesselate( mTess.get(), (int)winding, TESS_POLYGONS, 3, 2, 0 );
 	result->appendPositions( (vec2*)tessGetVertices( mTess.get() ), tessGetVertexCount( mTess.get() ) );
 	result->appendIndices( (uint32_t*)( tessGetElements( mTess.get() ) ), tessGetElementCount( mTess.get() ) * 3 );
-	
+
+	return result;
+}
+
+
+// * * *
+
+
+Triangulator3d::Triangulator3d( const PolyLine3f &polyLine )
+{
+	allocate();
+	addPolyLine( polyLine );
+}
+
+Triangulator3d::Triangulator3d()
+{
+	allocate();
+}
+
+void Triangulator3d::allocate()
+{
+	mAllocated = 0;
+
+	TESSalloc ma;
+	memset( &ma, 0, sizeof(ma) );
+	ma.memalloc = stdAlloc;
+	ma.memfree = stdFree;
+	ma.userData = (void*)&mAllocated;
+	ma.extraVertices = 2560; // realloc not provided, allow 256 extra vertices.
```

Comment:
Yeah been noticing some other weird behavior since that structure has an [`outOfMemory`](https://github.com/cinder/Cinder/blob/master/src/libtess2/tess.h#L53) member that isn't explicitly initialized to 0. So sporadically it seems to fail once it reuses memory the OS hasn't zeroed out.